store/sqlstore/post_store.go

```diff
@@ -865,106 +864,105 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
+		}
+
+		if params.OrTerms {
+			terms = strings.Join(strings.Fields(terms), " | ")
 		} else {
-			// no create date filters set
-			searchQuery = strings.Replace(searchQuery, "CREATEDATE_CLAUSE", "", 1)
+			terms = strings.Join(strings.Fields(terms), " & ")
 		}
 
-		if terms == "" {
-			// we've already confirmed that we have a channel or user to search for
-			searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", "", 1)
-		} else if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
-			// Parse text for wildcards
-			if wildcard, err := regexp.Compile(`\*($| )`); err == nil {
-				terms = wildcard.ReplaceAllLiteralString(terms, ":* ")
-			}
+		searchClause := fmt.Sprintf("AND %s @@  to_tsquery(:Terms)", searchType)
+		searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", searchClause, 1)
+	} else if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		searchClause := fmt.Sprintf("AND MATCH (%s) AGAINST (:Terms IN BOOLEAN MODE)", searchType)
+		searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", searchClause, 1)
 
-			if params.OrTerms {
-				terms = strings.Join(strings.Fields(terms), " | ")
-			} else {
-				terms = strings.Join(strings.Fields(terms), " & ")
+		if !params.OrTerms {
+			splitTerms := strings.Fields(terms)
+			for i, t := range strings.Fields(terms) {
+				splitTerms[i] = "+" + t
 			}
 
-			searchClause := fmt.Sprintf("AND %s @@  to_tsquery(:Terms)", searchType)
-			searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", searchClause, 1)
-		} else if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
-			searchClause := fmt.Sprintf("AND MATCH (%s) AGAINST (:Terms IN BOOLEAN MODE)", searchType)
-			searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", searchClause, 1)
-
-			if !params.OrTerms {
-				splitTerms := strings.Fields(terms)
-				for i, t := range strings.Fields(terms) {
-					splitTerms[i] = "+" + t
-				}
-
-				terms = strings.Join(splitTerms, " ")
-			}
+			terms = strings.Join(splitTerms, " ")
 		}
+	}
 
-		queryParams["Terms"] = terms
+	queryParams["Terms"] = terms
 
-		list := model.NewPostList()
+	list := model.NewPostList()
 
-		_, err := s.GetSearchReplica().Select(&posts, searchQuery, queryParams)
-		if err != nil {
-			mlog.Warn(fmt.Sprintf("Query error searching posts: %v", err.Error()))
-			// Don't return the error to the caller as it is of no use to the user. Instead return an empty set of search results.
-		} else {
-			for _, p := range posts {
-				if searchType == "Hashtags" {
-					exactMatch := false
-					for _, tag := range strings.Split(p.Hashtags, " ") {
-						if termMap[strings.ToUpper(tag)] {
-							exactMatch = true
-						}
-					}
-					if !exactMatch {
-						continue
+	_, err := s.GetSearchReplica().Select(&posts, searchQuery, queryParams)
+	if err != nil {
+		mlog.Warn(fmt.Sprintf("Query error searching posts: %v", err.Error()))
+		// Don't return the error to the caller as it is of no use to the user. Instead return an empty set of search results.
```

Comment:
Probably here we can return an empty set of results right away, and remove the else to keep the happy path at top level of the function.